open-craft/openedx-completion-aggregator
# YONK-1175: Fix performance issues with migration script.

**Description:** Changes query strategy to eliminate performance degradation.

**JIRA:** YONK-1175

**Dependencies:** N/A

**Merge deadline:** ASAP

**Installation instructions:** 

**Testing instructions:**

1.  Create 3 million stale completion records
2.  Run migrations
3.  Verify that logging output is created, similar to:
    
    ```
    Initial copy: Transferred records up to 100000 at 2018-10-19 18:21:50.099139+00:00
    Initial copy: Transferred records up to 200000 at 2018-10-19 18:21:53.551277+00:00
    Initial copy: Transferred records up to 300000 at 2018-10-19 18:21:57.221689+00:00
    Initial copy: Transferred records up to 400000 at 2018-10-19 18:22:00.731618+00:00
    Initial copy: Transferred records up to 500000 at 2018-10-19 18:22:04.137012+00:00
    Initial copy: Transferred records up to 600000 at 2018-10-19 18:22:07.616941+00:00
    ```
4. Verify that batches around 3,000,000 records in don't take longer than batches around 1,000,000 records in.   Note:  Because batches are determined by the value of the id, some batches, particularly near the beginning, have fewer records.

**Reviewers:**

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:**
